src/containerapp/azext_containerapp/custom.py

```diff
@@ -3258,3 +3260,54 @@ def set_registry_job(cmd, name, resource_group_name, server, username=None, pass
     containerapp_job_registry_set_decorator.construct_payload()
     r = containerapp_job_registry_set_decorator.set()
     return r
+
+
+# maintenance config
+def add_maintenance_config(cmd, resource_group_name, env_name, duration, start_hour_utc, weekday):
+    raw_parameters = locals()
+    maintenance_config_decorator = MaintenanceConfigPreviewDecorator(
+        cmd=cmd,
+        client=MaintenanceConfigPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    maintenance_config_decorator.construct_payload()
+    r = maintenance_config_decorator.add()
+    return r
+
+
+def update_maintenance_config(cmd, resource_group_name, env_name, duration, start_hour_utc, weekday):
+    raw_parameters = locals()
+    maintenance_config_decorator = MaintenanceConfigPreviewDecorator(
+        cmd=cmd,
+        client=MaintenanceConfigPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    maintenance_config_decorator.construct_payload()
+    r = maintenance_config_decorator.update()
+    return r
```

Comment:
What if customer only want to update one of the argument of the existing maintentance? 

I think you should handle `update` in `MaintenanceConfigPreviewUpdateDecorator`, `create` in `MaintenanceConfigPreviewCreateDecorator`.

Reply:
modified construct_payload to allow for create and update options